ImageInNavigationBarDemo/DemoTableViewController.swift

```diff
@@ -7,3 +7,157 @@
+        return 10
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "DemoCell", for: indexPath)
+        cell.textLabel?.text = "Sample"
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        performSegue(withIdentifier: "SegueID", sender: nil)
+    }
+
+    // MARK: - Delegates
+
+    override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard let height = navigationController?.navigationBar.frame.height else { return }
+        moveAndResizeImage2(for: height)
+    }
+
+    // MARK: - Private methods
+
+    private func moveAndResizeImage(for height: CGFloat) {
+        // Value of difference between icons for large and small states
+        let sizeDiff = Const.ImageSizeForLargeState * (1.0 - Const.SmallImageFactor) // 8.0
+
+        // This value = 14. It equals to difference of 12 and 6 (bottom spacing for large and small states)
+        // Also it adds 8.0 (size difference when the image gets smaller size)
+        let maxYTranslation = Const.ImageBottomSpaceForLargeState - Const.ImageBottomSpaceForSmallState + sizeDiff
+
+        if height <= Const.NavBarHeightSmallState {
+            // If NavBar height matches or smaller than Small state we tranform image to small state
+            imageView.transform = CGAffineTransform.identity
+                .scaledBy(x: Const.SmallImageFactor, y: Const.SmallImageFactor)
+                .translatedBy(x: sizeDiff, y: maxYTranslation)
+        } else if height >= Const.NavBarHeightLargeState {
+            // If NavBar height matches or bigger than Large state we tranform image to large state (basically to normal size)
+            imageView.transform = CGAffineTransform.identity
+                .scaledBy(x: 1.0, y: 1.0)
+                .translatedBy(x: 0.0, y: 0.0)
+        } else {
+            // If NavBar is in between of Small and Large states we tranform image and move bottom and right spacings proportionally to the size of NavBar. This way it creates a smooth animation for image resizing
+            let delta = height - Const.NavBarHeightSmallState
+            let coeff = delta / (Const.NavBarHeightLargeState - Const.NavBarHeightSmallState)
+            let sizeAddendumFactor = coeff * (1.0 - Const.SmallImageFactor)
+            let scale = sizeAddendumFactor + Const.SmallImageFactor
+
+            let yTranslation = maxYTranslation - (coeff * Const.ImageBottomSpaceForSmallState + coeff * sizeDiff)
+            let xTranslation = sizeDiff - coeff * sizeDiff
+
+            imageView.transform = CGAffineTransform.identity
+                .scaledBy(x: scale, y: scale)
+                .translatedBy(x: xTranslation, y: yTranslation)
+        }
+    }
+    
+    private func moveAndResizeImage2(for height: CGFloat) {
```

Comment:
I would structure grouped operations in this function for the better readability, i.e.:
```swift
let coeff: CGFloat = {
  let delta = height - Const.NavBarHeightSmallState
  let heightDifferenceBetweenStates = (Const.NavBarHeightLargeState - Const.NavBarHeightSmallState)
  return delta / heightDifferenceBetweenStates
}()

let scale: CGFloat = {
   ...
}()

let xTranslation: CGFloat = {
  ...
}()
...
```